lib/cusparse/generic.jl

```diff
@@ -339,6 +339,26 @@ function gemm!(transa::SparseChar, transb::SparseChar, alpha::Number, A::CuSpars
             handle(), transa, transb, Ref{T}(alpha), descA, descB, Ref{T}(beta),
             descC, T, algo, spgemm_desc, out, buffer)
     end
+    # cusparseSpMatGetSize retrieve the size of the output matrix and the number of non-zero elements
+    nnzC = Ref{Int64}()
+    cusparseSpMatGetSize(descC, Ref{Int64}(), Ref{Int64}(), nnzC)
+    # We suppose that AB and C have the same sparsity pattern if they have the same number of non-zero elements.
+    # Even if AB and C have the same number of non-zero elements they could have different
+    # sparsity patterns and the result will be wrong with β ≠ 0...
+    if nnz(C) ≠ nnzC[]
```

Comment:
Do we have a better way to compare sparsity patterns? Could be enabled in debug mode.

Reply:
To compare sparsity patterns, we need the sparsity pattern of `AB` and I think that it's as difficult and expensive as performing the product `AB`. :disappointed: